ci/docker.sh

```diff
@@ -45,23 +48,30 @@ if [ -n "${docker_tag}" ]; then
     .
 
   # Publishing to DockerHub
-  echo "" && echo "=== Publishing Docker image(s) on Docker Hub===" && echo ""
+  echo "" && echo "=== Publishing Docker image(s) on Docker Hub ===" && echo ""
   if [ -z "${DOCKER_USERNAME}" ] || [ -z "${DOCKER_PASSWORD}" ]; then
-    echo "Warning: DOCKER_USERNAME and/or DOCKER_USERNAME is(are) empty. Docker image is NOT going to be published on DockerHub !!!"
+    echo "Warning: DOCKER_USERNAME and/or DOCKER_PASSWORD variable(s) is(are) empty. Docker image(s) is(are) NOT going to be published on DockerHub !!!"
   else
     echo "${DOCKER_PASSWORD}" | docker login -u "${DOCKER_USERNAME}" --password-stdin
-    echo "Create tags"
     docker_images=("${evmapp_docker_image_name}" "${bootstraptool_docker_image_name}")
-    for docker_image in "${docker_images[@]}"; do
+
+    # Docker image(s) tags for PROD vs DEV release
+    if [ "${PROD_RELEASE}" = "true" ]; then
       tags=("${docker_tag}" "latest")
+    else
+      tags=("${docker_tag}" "dev")
+    fi
+
+    for docker_image in "${docker_images[@]}"; do
       for tag in "${tags[@]}"; do
+        echo "" && echo "Publishing docker image: ${docker_image}:${tag}"
         docker tag "${docker_image}:${docker_tag}" "index.docker.io/${docker_hub_org}/${docker_image}:${tag}"
```

Comment:
This would tag "${docker_image}:${docker_tag}" as ${docker_image}:${tag}" where docker_tag == tag. Probably a no-op, but not that elegant.

BTW, there is a way to tag an image already on the registry over the API, without having to reupload it or compare layer hashes. See https://github.com/HorizenOfficial/zen/blob/901e0cdbcbfa8bb0255bdc0eebd4e4717bb6f9a5/contrib/ci-horizen/scripts/prepare/docker_image_deploy.sh#L12-L24